pepr/robustness/foolbox_wrapper.py

```diff
@@ -0,0 +1,3230 @@
+                        result_tab.add_hline()
+                        result_tab.add_row(
+                            list(map(bold, ["", "Min. Rate", "Max. Rate"]))
+                        )
+                        result_tab.add_hline()
+                        result_tab.add_row(
+                            ["Epsilon", epsilons[worst_epsilon], epsilons[best_epsilon]]
+                        )
+                        result_tab.add_hline()
+                        result_tab.add_row(
+                            [
+                                "Success Rate",
+                                round(misclass[worst_epsilon], 3),
+                                round(misclass[best_epsilon], 3),
+                            ]
+                        )
+                        result_tab.add_hline()
+                        result_tab.add_row(
+                            [
+                                "Average L2 Distance",
+                                round(dist[worst_epsilon], 3),
+                                round(dist[best_epsilon], 3),
+                            ]
+                        )
+                        result_tab.add_hline()
+                else:
+                    with self.report_section.create(Tabular("|l|c|")) as result_tab:
+                        result_tab.add_hline()
+                        result_tab.add_row(["Epsilon", epsilons[0]])
+                        result_tab.add_hline()
+                        result_tab.add_row(
+                            [
+                                "Success Rate",
+                                round(misclass[0], 3),
+                            ]
+                        )
+                        result_tab.add_hline()
+                        result_tab.add_row(
+                            [
+                                "Average L2 Distance",
+                                round(dist[0], 3),
+                            ]
+                        )
+                        result_tab.add_hline()
+
+                self.report_section.append(Command("captionsetup", "labelformat=empty"))
+                self.report_section.append(
+                    Command("captionof", "table", extra_arguments="Attack Summary")
+                )
+
+
+class L2ContrastReductionAttack(BaseAttack):
+    """
+    foolbox.attacks.L2ContrastReductionAttack wrapper class.
+
+    Attack description:
+    Reduces the contrast of the input using a perturbation of the given size.
+
+    Parameters
+    ----------
+    attack_alias : str
+        Alias for a specific instantiation of the class.
+    attack_pars : dict
+        Dictionary containing all needed attack parameters:
+
+        * target (float): (optional) Target relative to the bounds from 0 (min) to 1
+          (max) towards which the contrast is reduced.
+        * epsilons (list): List of one or more Epsilons for the attack.
```

Comment:
Again **e**psilons?

Reply:
I think you wrote **E**psilons in the whole file?

Reply:
Maybe :joy: